qemu/tests/vioser_in_use.py

```diff
@@ -116,10 +133,20 @@ def run(test, params, env):
 
     bg_thread = run_bg_test(test, params, vm, sender)
     globals().get(params["interrupt_test"])(test, params, vm, session)
-    if bg_thread:
-        bg_thread.join(timeout=timeout, suppress_exception=suppress_exception)
-    if vm.is_alive():
-        kill_host_serial_pid(params, vm)
-        if (virtio_serial_file_transfer.transfer_data(params, vm, sender=sender) is
-                not True):
-            test.fail("Serial data transfter test failed.")
+
+    # for vcpu hotplug subtest, only check guest crash.
+    vcpu_hotplug = params.get_boolean("vcpu_hotplug")
+    if vcpu_hotplug:
+        error_context.context("Check if guest is alive.", logging.info)
+        vm.verify_kernel_crash()
+        session = vm.wait_for_login(timeout=timeout)
+        session.close()
```

Comment:
What do you think if `return` here and remove line 144? Thus narrow down the code hierarchy.

Reply:
The method is better than mine, it's ok for me, will update